fast64_internal/oot/oot_cutscene.py

```diff
@@ -0,0 +1,604 @@
+
+# Perhaps this should have been called something like OOTCSParentPropertyType,
+# but now it needs to keep the same name to not break existing scenes which use
+# the cutscene system.
+class OOTCSProperty():
+	propName = None
+	attrName = None
+	subprops = ["startFrame", "endFrame"]
+	expandTab : bpy.props.BoolProperty(default = True)
+	startFrame : bpy.props.IntProperty(name = '', default = 0, min = 0)
+	endFrame : bpy.props.IntProperty(name = '', default = 1, min = 0)
+		
+	def getName(self):
+		return self.propName
+		
+	def filterProp(self, name, listProp):
+		return True
+	
+	def filterName(self, name, listProp):
+		return name
+	
+	def draw(self, layout, listProp, listIndex, cmdIndex, objName, collectionType):
+		layout.prop(self, 'expandTab', text = self.getName() + " " + str(cmdIndex),
+			icon = 'TRIA_DOWN' if self.expandTab else 'TRIA_RIGHT')
+		if not self.expandTab: return
+		box = layout.box().column()
+		drawCollectionOps(box, cmdIndex, collectionType + "." + self.attrName,
+			listIndex, objName)
+		for p in self.subprops:
+			if self.filterProp(p, listProp):
+				prop_split(box, self, p, self.filterName(p, listProp))
+
+class OOTCSTextboxProperty(OOTCSProperty, bpy.types.PropertyGroup):
+	propName = "Textbox"
+	attrName = "textbox"
+	subprops = ["messageId", "ocarinaSongAction", "startFrame", "endFrame",
+		"type", "topOptionBranch", "bottomOptionBranch", "ocarinaMessageId"]
+	textboxType : bpy.props.EnumProperty(items = ootEnumCSTextboxType)
+	messageId : bpy.props.StringProperty(name = '', default = '0x0000')
+	ocarinaSongAction : bpy.props.StringProperty(name = '', default = '0x0000')
+	type : bpy.props.StringProperty(name = '', default = '0x0000')
+	topOptionBranch : bpy.props.StringProperty(name = '', default = '0x0000')
+	bottomOptionBranch : bpy.props.StringProperty(name = '', default = '0x0000')
+	ocarinaMessageId : bpy.props.StringProperty(name = '', default = '0x0000')
+	
+	def getName(self):
+		return self.textboxType
+	
+	def filterProp(self, name, listProp):
+		if self.textboxType == "Text":
+			return name not in ["ocarinaSongAction", "ocarinaMessageId"]
+		elif self.textboxType == "None":
+			return name in ["startFrame", "endFrame"]
+		elif self.textboxType == "LearnSong":
+			return name in ["ocarinaSongAction", "startFrame", "endFrame", "ocarinaMessageId"]
+		else:
+			raise PluginError("Invalid property name for OOTCSTextboxProperty")
+
+class OOTCSTextboxAdd(bpy.types.Operator):
+	bl_idname = 'object.oot_cstextbox_add'
+	bl_label = 'Add CS Textbox'
+	bl_options = {'REGISTER', 'UNDO'} 
+
+	collectionType : bpy.props.StringProperty()
+	textboxType : bpy.props.EnumProperty(items = ootEnumCSTextboxType)
+	listIndex : bpy.props.IntProperty()
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		collection = getCollection(self.objName, self.collectionType, self.listIndex)
+		collection.add()
+		collection[len(collection)-1].textboxType = self.textboxType
+		#self.report({'INFO'}, 'Success!')
```

Comment:
there are a bunch of commented out lines starting from this line, I think most of them can go?

Reply:
Changed. The commented out lines below that were for unknown or unused properties. These are properties that are in the cutscene data, and sometimes set to nonzero values that seem to have meaning, but they have been confirmed by decomp (at least in OoT) to never be read by the game. I left them as commented out because if you wanted to create arbitrary *cutscene data*--not merely create a specific behavior in a cutscene in the game--these could be uncommented and used. However, I didn't actually have commented-out support for them through the exporting process, and I doubt anyone is going to care what's in the unused data, so I just removed them.